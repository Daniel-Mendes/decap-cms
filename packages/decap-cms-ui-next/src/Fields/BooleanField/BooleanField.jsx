import React from 'react';

import Field from '../../Field';
import Switch from '../../Switch';

function BooleanField({ name, onChange, children, value, ...props }) {
  return (
    <Field
      {...props}
      labelTarget={name}
      control
      onClick={() => (value ? onChange(false) : onChange(true))}
    >
      <Switch
        onChange={() => (value && onChange ? onChange(false) : onChange(true))}
        checked={value}
      />
      {children && children}
    </Field>
  );
}

export default BooleanField;
