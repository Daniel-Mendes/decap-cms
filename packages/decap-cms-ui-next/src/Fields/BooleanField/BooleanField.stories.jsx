import React from 'react';
import styled from '@emotion/styled';
import { useArgs } from '@storybook/preview-api';

import BooleanField from '.';

const StyledBooleanField = styled(BooleanField)`
  width: 33vw;
`;

export default {
  title: 'Fields/BooleanField',
  component: BooleanField,
  argTypes: {
    errors: {
      control: 'object',
      if: {
        arg: 'error',
      },
    },
  },
  args: {
    label: 'Boolean Field Label',
    description: 'This is a description',
    status: 'Required',
    value: false,
    inline: false,
    error: false,
    errors: [
      {
        message: 'Error message.',
      },
    ],
  },
};

export function _BooleanField(args) {
  const [{ value }, updateArgs] = useArgs();

  function toggleValue() {
    updateArgs({ value: !value });
  }

  return <StyledBooleanField {...args} value={value} onChange={toggleValue} />;
}
