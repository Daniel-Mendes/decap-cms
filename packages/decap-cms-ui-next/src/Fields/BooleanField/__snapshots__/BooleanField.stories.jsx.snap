// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fields/BooleanField _BooleanField smoke-test 1`] = `"<div class="emi8uzw0 css-1l4uu3i-FieldWrap-StyledBooleanField e1x0ipyd6"><div class="css-1uqy5pq-FieldInside-FieldInside-FieldInside-FieldInside e1x0ipyd5"><label class="css-1n9k11u-Label-StyledLabel e1x0ipyd3">Boolean Field Label <div color="neutral" radius="large" class="css-6930ng-BadgeWrap-BadgeWrap epfx4ki3">Required</div></label><div class="css-dj548v-ChildrenWrap e1x0ipyd4"><input type="checkbox" class="css-1rg9wi1-StyledInput e1lye9la1"><button class="css-1407tdr-SwitchButton e1lye9la0"></button></div><p class="css-nalsj0-StyledDescription e1x0ipyd1">This is a description</p></div></div>"`;
